src/services/codefixes/importFixes.ts

```diff
@@ -644,6 +665,19 @@ namespace ts.codefix {
             Debug.fail("Either the symbol or the JSX namespace should be a UMD global if we got here");
         }
 
+        const { module, allowSyntheticDefaultImports } = context.compilerOptions;
+
+        // Prefer to import as a synthetic `default` if available.
+        if (allowSyntheticDefaultImports || module === ModuleKind.System && allowSyntheticDefaultImports !== false) {
+            return getCodeActionForImport(symbol, { ...context, symbolName, kind: ImportKind.Default });
+        }
+
+        // When a synthetic `default` is unavailable, use `import..require` if the module kind supports it.
+        if (module === ModuleKind.AMD || module === ModuleKind.CommonJS || module === ModuleKind.UMD) {

```

Comment:
not sure i agree we need this change.

Reply:
The argument here is that if `require` is available, it is a reliable away to obtain `module.exports` even when it is callable. I am happy to remove this as requested but believe it is valuable.

Reply:
This change is actually what's causing the build failures because of expectations in the four slash tests.

Would you like me to remove the `require` changes? I'd be happy to open a separate issue for them.